Renderer.swift

```diff
@@ -103,11 +104,16 @@ class Renderer: NSObject {
                 let projectionMatrix = ortho(left: 0, right: 200 * aspectRatio, bottom: 0, top: 200, near: -1, far: 10)
                 var uniforms = Uniforms(modelViewMatrix: mvMatrix, projectionMatrix: projectionMatrix)
                 
+                // REVIEW: I would look for a way that you can change the uniform values without having to re-create
+                // the whole command buffer every time.
                 commandEncoder.setVertexBytes(&uniforms, length: MemoryLayout<Uniforms>.size, index: 1)
                 
                 commandEncoder.setRenderPipelineState(renderPipeline)
                 commandEncoder.setCullMode(MTLCullMode.back)
                 
+                // REVIEW: This, in particular, seems like it would be pretty expensive, at least algorithmically.
+                // The crabs are probably not ACTUALLY that complex, but a double-nested 'for' loop inside every frame
+                // for drawing something shouldn't be necessary... like I would be surprised if this is actually needed.
```

Comment:
Even though this is a nested 'for', you are right and the crab model is pretty simple. I think there are just two meshes: front and back. So while the algorithm is n^2, for this case, assuming the compiler is smart enough, it'll get unrolled into running only four times.